### DISCUSSION TITLE
What is Code Security?

### DISCUSSION THREAD

**Discussion by iLakshya** [community/community#33629] [Code Security]
Quick question...

What is Code Security? Like why do we have to use code security? It's an open-source platform, right? And If you have made your repository public, then why do you have to add code security? Also if your repository is private, then also why is code security required Please explain this to me. 

Also, please give me your thoughts on the same.

**Comment by Murat-Karakaya** [comment]
Code Security: Is simply making your code strong against attacks. So your website wont get hacked by a hacker.

Public repositories need security because when you write code, everyone can see your code. So don't put your "api key" or "ssh key" into your puplic repository. And don't trust users. don't allow them to sign in without using their password. Stuff like that.

Private repositories need code security. Because making them private, doesn't fixes everytihng. A private repository can be vulnerable to attacks like SQL injections, DDOS attacks, CRSF attacks and more...

Luckly, there are great free tutorials out there theaching how to be prepared against all these dangers.


I hope I understood your question correctly  : )
